packages/central-server/src/apiV2/projects/CreateProject.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { CreateHandler } from '../CreateHandler';
+import { assertAnyPermissions, assertBESAdminAccess } from '../../permissions';
+
+/**
+ * Handles POST endpoints:
+ * - /projects
+ */
+
+export class CreateProject extends CreateHandler {
+  async assertUserHasAccess() {
+    await this.assertPermissions(
+      assertAnyPermissions([assertBESAdminAccess], 'You need BES Admin to create new projects'),
+    );
+  }
+
+  async createRecord() {
+    const {
+      code,
+      name,
+      description,
+      sort_order,
+      image_url,
+      logo_url,
+      permission_groups,
+      countries,
+      entityTypes,
+    } = this.newRecordData;
+
+    await this.createProjectEntity(code, name);
```

Comment:
It's cool that we do all this together! Maybe just need to wrap these in a transaction block so if one fails they all rollback.

Here's an example https://github.com/beyondessential/tupaia/blob/dev/packages/central-server/src/apiV2/import/importDisaster.js#L53

Reply:
@IgorNadj ah yeah, thats a great point. I was having this issue when testing locally.